File: go-libs/types/workout.go
package types

import "gorm.io/gorm"

// Workout is for a collection of exercises
type Workout struct {
	gorm.Model
	ID        string       `json:"id" gorm:"primaryKey"`
	Execises  [][]Exercise `json:"execises"`
	StartTime int64        `json:"start_time"`
	EndTime   int64        `json:"end_time"`
	Notes     string       `json:"notes"`
	Title     string       `json:"title"`
}

File: app/src/types/workout.ts
import { Exercise } from "./exercise"

export type Workout = {
    workout_id: string,
    // user_id:string,
    start_time: string,
    end_time: string,
    notes: string,
    title: string,
}

export type WorkoutGroup = {
    workout: Workout,
    exercises:Exercise[],
}